plugins/modules/consul_role.py

```diff
@@ -0,0 +1,491 @@
+    required: true
+    type: str
+  state:
+    description:
+      - whether the role should be present or absent.
+    required: false
+    choices: ['present', 'absent']
+    default: present
+    type: str
+  description:
+    description:
+      - Description of the role.
+    required: false
+    type: str
+    default: ''
+  policies:
+    type: list
+    elements: dict
+    description:
+      - List of policies to attach to the role.
+      - Each element must have a "name" or "id" (or both) to identify the policy. See consul_policy for more info.
+    required: false
+    default: []
+  service_identities:
+    type: list
+    elements: dict
+    description:
+      - List of service identities to attach to the role.
+      - Each element must have a "name" and optionally a "datacenters" list of datacenters the policy is valid for.
+      - An empty datacenters list allows all datacenters
+    required: false
+    default: []
+  node_identities:
+    type: list
+    elements: dict
+    description:
+      - List of node identities to attach to the role.
+      - Each element must have a "name" and optionally a "datacenter" the policy is valid for. An empty datacenter allows all datacenters
+    required: false
+    default: []
+  host:
+    description:
+      - host of the consul agent, defaults to localhost.
+    required: false
+    default: localhost
+    type: str
+  port:
+    type: int
+    description:
+      - the port on which the consul agent is running.
+    required: false
+    default: 8500
+  scheme:
+    description:
+      - the protocol scheme on which the consul agent is running.
+    required: false
+    default: http
+    type: str
+  token:
+    description:
+      - a management token is required to manipulate the roles.
+    type: str
+  validate_certs:
+    type: bool
+    description:
+      - whether to verify the tls certificate of the consul agent
+    required: false
+    default: True
+requirements:
+  - requests
+'''
+
+EXAMPLES = """
+- name: Create a role with 2 policies
+  community.general.consul_role:
+    host: consul1.example.com
+    token: some_management_acl
+    name: foo-role
+    policies:
+      - id: 783beef3-783f-f41f-7422-7087dc272765
+      - name: "policy-1"
+- name: Create a role with service identity
+  community.general.consul_role:
+    host: consul1.example.com
+    token: some_management_acl
+    name: foo-role-2
+    service_identities:
+      - name: web
+        datacenters:
+          - dc1
+- name: Create a role with node identity
+  community.general.consul_role:
+    host: consul1.example.com
+    token: some_management_acl
+    name: foo-role-3
+    node_identities:
+      - name: node-1
+        datacenter: dc2
+- name: Remove a role
+  community.general.consul_role:
+    host: consul1.example.com
+    token: some_management_acl
+    name: foo-role-3
+    state: absent
+"""
+
+RETURN = """
+role:
+    description: The role object
+    returned: success
+    type: str
```

Comment:
Shouldn't the return value be a JSON object (i.e. `type: dict`) instead of a string?